src/nvim/ex_getln.c

```diff
@@ -2393,10 +2394,11 @@ static int command_line_changed(CommandLineState *s)
       && !vpeekc_any()
       && cmdpreview_may_show(s)) {
     // 'inccommand' preview has been shown.
-  } else if (cmdpreview) {
-    cmdpreview = false;
-    update_screen(UPD_SOME_VALID);  // Clear 'inccommand' preview.
   } else {
+    cmdpreview = false;
```

Comment:
this is a bit subtle, a small comment like this may help:

    cmdpreview = false;  // cmdpreview_may_show() returned false: preview just ended.

Reply:
I cannot think of a 100% accurate comment to describe this.